### DISCUSSION TITLE
What is the current level of support for PyTorch?

### DISCUSSION THREAD

**Discussion by dibgerge** [awslabs/gluonts#1302] [Q&A] [answered]
The documentation does not mention PyTorch, but I see that there is active work of adding support for it. 
Is there currently any release that supports Pytorch or it is currently only on master? 
In any case, what is the current level of support? Can we run any of the models with Pytorch, or a subset of them? In particular, I am interested in DNN, DeepAR and gpvar models.

Thanks!

**Comment by lostella** [comment, marked as answer]
Hi @dibgerge PyTorch support is currently under development and lives on`master`: there we have added some components to help model development (e.g. the `DistributionOutput` classes), added support to PyTorch tensors to the data loader class, support for `Predictor` types wrapping PyTorch networks. An example of how to use all of this is [in a test](https://github.com/awslabs/gluon-ts/blob/master/test/torch/model/test_simple_torch_model.py) and will be included among the example notebooks, see #1289.

We don't have yet plans of translating existing MXNet models to PyTorch, mainly because of the potential overhead of having to maintain/sync the two implementations. New models may be implemented in PyTorch, though.
